lapis/utility/monitor.py

```diff
@@ -1,19 +1,40 @@
+import copy
 from typing import Callable, TYPE_CHECKING
 
 import logging
+import logging.handlers
 
-from usim import each, Flag
-
-from lapis.cost import cobald_cost
+from cobald.monitor.format_json import JsonFormatter
+from cobald.monitor.format_line import LineProtocolFormatter
+from usim import each, Flag, time
 
 if TYPE_CHECKING:
     from lapis.simulator import Simulator
 
 sampling_required = Flag()
 
 
+class LoggingSocketHandler(logging.handlers.SocketHandler):
+    def makePickle(self, record):
+        return self.format(record).encode()
+
+
+class LoggingUDPSocketHandler(logging.handlers.DatagramHandler):
+    def makePickle(self, record):
+        return self.format(record).encode()
+
+
+class TimeFilter(logging.Filter):
```

Comment:
The class name seems misleading - it is not actually working with time. ``SimulationTimeFilter`` may be more appropriate.

You might want to drop ``Filter`` from the name completely, but it is a subjective choice between what the class does versus how it does it.

Reply:
Renamed it to `SimulationTimeFilter`.